pkg/loki/modules.go

```diff
@@ -740,11 +745,30 @@ func (t *Loki) initIndexGateway() (services.Service, error) {
 		return nil, err
 	}
 
-	gateway := indexgateway.NewIndexGateway(shipperIndexClient)
+	gateway, err := indexgateway.NewIndexGateway(t.Cfg.IndexGateway, util_log.Logger, prometheus.DefaultRegisterer, shipperIndexClient.(*shipper.Shipper), shipperIndexClient)
+	if err != nil {
+		return nil, err
+	}
+
+	t.Server.HTTP.Path("/indexgateway/ring").Methods("GET", "POST").Handler(gateway)
 	indexgatewaypb.RegisterIndexGatewayServer(t.Server.GRPC, gateway)
 	return gateway, nil
 }
 
+func (t *Loki) initIndexGatewayRing() (_ services.Service, err error) {
```

Comment:
I think we should have a check for whether the index gateway mode is simple and just return if that is the case.

Reply:
Index gateway is optional to run and it is possible that Loki would break setting up unexpected things.

Reply:
Ooops I overlooked this one, I thought this was the `IndexGateway` service but it is actually the IndexGatewayRing. I added a guard clause as you suggested to not start this if mode != ring.

Reply:
Looks good. Thanks!